### ISSUE TITLE
Building pdf.js using gulp generice does not create the files wanted

### ISSUE THREAD

**Issue by ziracmo** [mozilla/pdf.js#9818]
Hi everyone,

I'm doing a fork of the project pdf.js to be able to print the annotations from a home made library (highlight, stamp, comment).
However, when i try to build the project after i did some changes using the command _gulp generic_, and i replace the _viewer.js_ created during the build by the old one. And when do that i have two errors :

- The first one :

![pdfjs-disablecaptureerror](https://user-images.githubusercontent.com/12600085/41473154-b82ff956-70b8-11e8-9aa4-984c7b08ce9a.PNG)

- The second one :

![pdfjs-notdefinedpdfviewerapp](https://user-images.githubusercontent.com/12600085/41473180-d072cdae-70b8-11e8-9acb-bd217fb265fa.PNG)

For the second one, i think the variable PDFViewerApplication is not creating by the build from _gulp generic_  in the _viewer.js_, and i would like to know if there is any way to do it or i'm missing something

I also tried this about the pdf.js file and see if the variable PDFJS is correctly created, but i only have a pdfjsLib unlike before where i hade PDFJS.

Configuration:
- Chrome 67.0
- Operating system and its version: Windows 10

Thanks for your help, don't hesitate to ask more informations

**Comment by timvandermeij** [comment]
It looks like you're using PDF.js version 2.0. A lot has changed for version 2.0, among which the removal of the `PDFJS` global. It doesn't work to just place back an old `viewer.js` file (most of the logic is now in `app.js` by the way). For example, `disableCreateObjectURL` is now a proper app option (preference), refer to: https://github.com/mozilla/pdf.js/blob/73843865b88806f7c4a12b7bf319fd87e7df761a/web/app_options.js#L148-L152

In short, if you do an upgrade of PDF.js, you cannot do it partially. You have to integrate your custom changes in the viewer belonging to the PDF.js version you're using.
